loopchain/blockchain/blocks/v1_0/block_verifier.py

```diff
@@ -1,25 +1,106 @@
+    from loopchain.blockchain.invoke_result import InvokePool
+    from loopchain.blockchain.blocks.v1_0.block import Block, BlockHeader
 
 
 class BlockVerifier(DataVerifier):
     version = "1.0"
 
-    def __init__(self, tx_versioner: TransactionVersioner, invoke_pool: 'InvokePool'):
+    def __init__(self, tx_versioner: TransactionVersioner, invoke_pool: 'InvokePool', reps_getter):
         self._invoke_pool: 'InvokePool' = invoke_pool
         self._tx_versioner: TransactionVersioner = tx_versioner
+        self._reps_getter = reps_getter
+
+    async def verify(self, prev_block: 'Block', block: 'Block'):
+        self._verify_version(block)
+        self._verify_common(block)
+
+        if block.header.height > 1:
+            self._verify_prev_votes(prev_block, block)
+
+        self._verify_transactions(block)
+        self._verify_signature(block)
+        self._invoke_pool.invoke(block, self._tx_versioner)
+
+    def _verify_version(self, block: 'Block'):
+        if block.header.version != self.version:
+            raise BlockVersionNotMatch(block.header.version, self.version,
+                                       f"The block version is incorrect. Block({block.header})")
+
+    def _verify_common(self, block: 'Block'):
+        header: 'BlockHeader' = block.header
+
+        if header.timestamp is None:
+            raise RuntimeError(f"Block({header.height}, {header.hash.hex()} does not have timestamp.")
+        if header.height > 0 and header.prev_hash is None:
+            raise RuntimeError(f"Block({header.height}, {header.hash.hex()} does not have prev_hash.")
+
+    def _verify_prev_votes(self, prev_block: 'Block', block: 'Block'):
+        prev_votes: 'BlockBody' = block.body.prev_votes
+        for vote in prev_votes:
+            # Check for only a vote made by `Siever`.
+            if not vote:
+                continue
+
+            if vote.version <= '0.5':
+                header: 'BlockHeader' = block.header
+                prev_next_reps_hash = prev_block.header.revealed_next_reps_hash
+                if prev_next_reps_hash:
+                    if prev_next_reps_hash != header.validators_hash:
+                        raise RuntimeError(
+                            f"Block({header.height}, {header.hash.hex()}, "
+                            f"RepsHash({header.reps_hash}), "
+                            f"Expected({prev_next_reps_hash}).")
+                    prev_reps = self._reps_getter(prev_block.header.reps_hash)
+                else:
+                    prev_reps = self._reps_getter(header.reps_hash)
+
+                return self._verify_prev_votes_by_siever(block, prev_reps, vote)
+
+            vote.verify()
+
+    def _verify_prev_votes_by_siever(self, block: 'Block', prev_reps: Sequence['ExternalAddress'], any_vote):
+        header: 'BlockHeader' = block.header
+        body: 'BlockBody' = block.body
+        votes_class = v0_5.BlockVotes if any_vote.version else v0_1a.BlockVotes
```

Comment:
class `Votes` provides useful method called `get_block_votes_class`. How about using it?
This helps avoiding import lines of Votes `v0_1a, v0_5`.